lib/active_model_serializers/adapter/json_api/pagination_links.rb

```diff
@@ -53,11 +53,16 @@ def last_page_url
 
         def prev_page_url
           return nil if collection.current_page == FIRST_PAGE
-          url_for_page(collection.current_page - FIRST_PAGE)
+
+          if collection.current_page > collection.total_pages
+            url_for_page(collection.total_pages)
+          else
+            url_for_page(collection.current_page - FIRST_PAGE)
+          end
```

Comment:
of

```diff
-          url_for_page(collection.current_page - FIRST_PAGE)
+          page_number = collection.current_page > collection.total_pages ? collection.total_pages : collection.current_page - FIRST_PAGE
+          url_for_page(page_number)
```

So, `prev` is the last page if `current page` is out of bounds,

Reply:
Done.